testing-and-tdd/testing-and-tdd.md

```diff
@@ -107,12 +107,12 @@ But what if your `class` uses another `class`? And what if your `function` calls
   - Since it's all isolated it shouldn't suffer from weird effects.
   - You can even mock the system clock to remove time dependencies! :-O
 - Realism: **low**.
-  - It's been said unit tests test that the code does what the developer intended, not what the user necessarily wants.
+  - It's been said that a 'unit test' tests that the code does what the developer intended - not necessarily what the user wants.
   - Just because your `SuperDuperStringUtils` works doesn't mean its usage in your login page to, say, format user's custom gender will do what you _really_ want.
 
 #### Integration Tests
 
-This term does not have much consensus on its meaning, but it generally refers to tests that test two (or more) units working together. It could be two classes/objects or a function that calls another function. It can also having your API call another real API, which is quite different (and slower) than a unit test. But some of the ideas are the same.
+This term does not have much consensus on its meaning, but it generally refers to tests that test two (or more) units working together. It could be two classes/objects or a function that calls another function. It can also be of having your API call another real API, which is quite different (and slower) than a unit test. But some of the ideas are the same.
```

Comment:
It might be clearer to change:

> It can also be of having your API call another real API, which is quite different (and slower) than a unit test.

to:

It can also mean having your API call another real API, which is quite different from (and slower than) a unit test.